### PR TITLE
[enhancement](broker) broker load support tencent cos

### PR DESCRIPTION
**Pull request by nextdreamblue**
Signed-off-by: nextdreamblue <zxw520blue1@163.com>

# Proposed changes

Issue Number: close #xxx

## Problem summary

 broker load support tencent cos.

```
使用腾讯cos对象存储的协议去访问cos相比原生s3a协议的优势：

比如针对对象存储的数据读取，又进行数据的预读操作
针对文件路径的查询操作，在判断目录是否是目录时，有进行list-1的优化操作
支持了对象存储这边的融合桶支持
支持了ranger的访问控制协议 
其他兼容性优化
```

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] No
3. Has unit tests been added:
    - [ ] No Need
4. Has document been added or modified:
    - [ ] Yes
5. Does it need to update dependencies:
    - [ ] No
6. Are there any changes that cannot be rolled back:
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
Why not access cos with s3?

**Comment by nextdreamblue** [comment]
> Why not access cos with s3?

使用腾讯cos对象存储的协议去访问cos相比原生s3a协议的优势：
1. 比如针对对象存储的数据读取，又进行数据的预读操作
2. 针对文件路径的查询操作，在判断目录是否是目录时，有进行list-1的优化操作
3. 支持了对象存储这边的融合桶支持
4. 支持了ranger的访问控制协议 
5. 其他兼容性优化

**Comment by nextdreamblue** [comment]
@morningman please review, thanks

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 39.65 seconds
 load time: 572 seconds
 storage size: 17154655360 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221028101950_clickbench_pr_35283.html

File: fs_brokers/apache_hdfs_broker/build.sh [hunk 1]
```diff
@@ -41,6 +41,10 @@ rm -rf "${BROKER_OUTPUT}"
 install -d "${BROKER_OUTPUT}/bin" "${BROKER_OUTPUT}/conf" \
     "${BROKER_OUTPUT}/lib"
 
+# download tencent hadoop cos jar
+wget https://github.com/tencentyun/hadoop-cos/releases/download/v8.1.7/hadoop-cos-2.7.5-8.1.7.jar
+mv hadoop-cos-2.7.5-8.1.7.jar "${BROKER_OUTPUT}/lib/"
+
```
Comment (lide-reed) [inline fs_brokers/apache_hdfs_broker/build.sh:47]: Need to make sure if the downloading is necessary or using maven lib.

Comment (nextdreamblue) [reply fs_brokers/apache_hdfs_broker/build.sh:47]: 1.目前 Hadoop-COS 已经正式被 Apache Hadoop-3.3.0。但是目前doris依赖的hadoop版本比较低，无法使用该版本；
2.在 Apache Hadoop-3.3.0，需要下载单独 Hadoop-cos jar 包后。
3.目前单独的Hadoop-cos jar 包，腾讯云未提交到maven，因此目前只能使用单独下载的方式

后续如果hadoop库升级或者支持maven方式下载，再调整此处的下载方式

**Review by lide-reed — DISMISSED** [review]
LGTM

**Review by lide-reed — DISMISSED** [review]
LGTM

**Review by lide-reed — APPROVED** [review]
LGTM

**Pull request merged**